collection-scripts/gather

```diff
@@ -1,24 +1,65 @@
 #!/bin/bash
 
+source "common.sh"
+
+# for new flag to get since and since-time
+get_log_collection_args
+
+
 # generate /must-gather/version file
-. version
 echo "opendatahub/must-gather" > /must-gather/version
 version >> /must-gather/version
 
+# define default namespaces
 OPERATOR_NS=${OPERATOR_NAMESPACE:-redhat-ods-operator}
 NOTEBOOKS_NS=${NOTEBOOK_NAMESPACE:-rhods-notebooks}
 MONITORING_NS=${MONITORING_NAMESPACE:-redhat-ods-monitoring}
 APPLICATIONS_NS=${APPLICATIONS_NAMESPACE:-redhat-ods-applications}
 
-oc adm inspect namespace/$OPERATOR_NS --dest-dir=must-gather || echo "Error getting logs from ${OPERATOR_NS}"
-oc adm inspect namespace/$NOTEBOOKS_NS --dest-dir=must-gather || echo "Error getting logs from ${NOTEBOOKS_NS}"
-oc adm inspect namespace/$MONITORING_NS --dest-dir=must-gather  || echo "Error getting logs from ${MONITORING_NS}"
-oc adm inspect namespace/$APPLICATIONS_NS --dest-dir=must-gather  || echo "Error getting logs from ${APPLICATIONS_NS}"
+oc adm inspect ${log_collection_args} namespace/$OPERATOR_NS --dest-dir="$DST_DIR" || echo "Error getting logs from ${OPERATOR_NS}"
+oc adm inspect ${log_collection_args} namespace/$NOTEBOOKS_NS --dest-dir="$DST_DIR" || echo "Error getting logs from ${NOTEBOOKS_NS}"
+oc adm inspect ${log_collection_args} namespace/$MONITORING_NS --dest-dir="$DST_DIR"  || echo "Error getting logs from ${MONITORING_NS}"
+oc adm inspect ${log_collection_args} namespace/$APPLICATIONS_NS --dest-dir="$DST_DIR"  || echo "Error getting logs from ${APPLICATIONS_NS}"
 
-/usr/bin/gather_data_science_pipelines
-/usr/bin/gather_model_mesh
-/usr/bin/gather_notebooks
-# force disk flush to ensure that all data gathered is accessible in the copy container
-sync
+# add DSCI and DSC
+get_operator_resource dscinitialization
+get_operator_resource datasciencecluster
 
+# set component or default to all
+component=${COMPONENT:-all}
+case "$component" in
+    "dsp")
+        /usr/bin/gather_data_science_pipelines
+        ;;
+    "modelmesh")
+        /usr/bin/gather_serving
+        ;;
+    "kserve")
+        /usr/bin/gather_serving
+        ;;
+    "dashboard")
+        echo "It has been included in the default namespace."
+        ;;
+    "workbench")
+        /usr/bin/gather_notebooks
+        ;;
+    "dw")
+        /usr/bin/gather_codeflare
+        /usr/bin/gather_kuberay
+        /usr/bin/gather_kueue
+        ;;
+    "mr"
+        echo "TBD."
+        ;;
+    *) # for all 
+        /usr/bin/gather_data_science_pipelines
+        /usr/bin/gather_serving
+        /usr/bin/gather_notebooks
+        /usr/bin/gather_codeflare
+        /usr/bin/gather_kuberay
+        /usr/bin/gather_kueue
+        ;;
```

Comment:
Minor comment, this is a bit error prone: every time we add a new component/command we need to add it under the component _and_ the "all" branch

Reply:
yep. 
i was thinking about this when i added the "COMPONENT" part:  maybe we should add "update must-gather as one step in the new component onboarding list"?
or we need a new design to struct these functions for new components in a better way.